docs/howto.rst

```diff
@@ -0,0 +1,1147 @@
+   [ReFrame Setup]
+     version:           4.0.0-dev.1
+     command:           '/code/bin/reframe -c tutorials/flux -C tutorials/flux/settings.py --run'
+     launched by:       root@b1f6650222bc
+     working directory: '/code'
+     settings file:     'tutorials/flux/settings.py'
+     check search path: '/code/tutorials/flux'
+     stage directory:   '/code/stage'
+     output directory:  '/code/output'
+
+   [==========] Running 1 check(s)
+   [==========] Started on Fri Sep 16 20:47:15 2022
+
+   [----------] start processing checks
+   [ RUN      ] EchoRandTest /66b93401 @generic:default+builtin
+   [       OK ] (1/1) EchoRandTest /66b93401 @generic:default+builtin
+   [----------] all spawned checks have finished
+
+   [  PASSED  ] Ran 1/1 test case(s) from 1 check(s) (0 failure(s), 0 skipped)
+   [==========] Finished on Fri Sep 16 20:47:15 2022
+   Run report saved in '/root/.reframe/reports/run-report.json'
+   Log file(s) saved in '/tmp/rfm-0avso9nb.log'
+
+For advanced users or developers, here is how to run tests within the container:
+
+Testing
+-------
+
+.. code-block:: console
+
+    ./test_reframe.py --rfm-user-config=tutorials/flux/settings.py unittests/test_schedulers.py -xs
+
+
+Building test libraries and utilities
+=====================================
+
+ReFrame tests are extremely modular.
+You can create libraries of base tests and utilities that others can use or extend.
+You can organize the source code of a test library as you would with a regular Python code.
+Let's see a made-up example for demonstration purposes:
+
+.. code-block:: console
+
+    ~/reframe-examples/howto
+    ├── testlib
+    │   ├── __init__.py
+    │   ├── simple.py
+    │   └── utility
+    │       └── __init__.py
+    └── testlib_example.py
+
+
+The ``testlib_example.py`` is fairly simple:
+it extends the :class:`simple_echo_check` from the test library and sets the message.
+
+.. literalinclude:: ../examples/howto/testlib_example.py
+   :caption:
+   :lines: 6-
+
+The :class:`simple_echo_check` it echoes "Hello, <message>" and asserts the output.
+It also uses a dummy fixture that it includes from a utility.
+
+.. literalinclude:: ../examples/howto/testlib/simple.py
+   :caption:
+   :lines: 6-
+
+Note that the :class:`simple_echo_check` is also decorated as a :func:`@simple_test <reframe.core.decorators.simple_test>`, meaning that it can be executed as a stand-alone check.
+This is typical when you are building test libraries:
+you want the base tests to be complete and functional making minimum assumptions for the target system/environment.
+You can then specialize further the derived tests and add more constraints in their :attr:`valid_systems` or :attr:`valid_prog_environs`.
+
+Let's try running both the library and the derived tests:
+
+.. code-block:: bash
+   :caption: Running the derived test
+
+    reframe -c reframe-examples/howto/testlib_example.py -r
```

Comment:
Adapt this for the workdir change in docker.

Reply:
Where? We pass the `--workdir` to the EB-Spack container. For the rest of the tutorial we were doing `cd ...`